src/interactions/slashCommands/mixins/CommandOptionWithChoices.ts

```diff
@@ -46,17 +51,45 @@ export abstract class ApplicationCommandOptionWithChoicesBase<T extends string |
 	 *
 	 * @param choices The choices to add
 	 */
-	public addChoices(choices: [name: string, value: T][]) {
+	public addChoices(choices: [name: string, value: T][]): Omit<this, 'setAutocomplete'> {
+		if (this.autocomplete) {
+			throw new RangeError('Autocomplete and choices are mutually exclusive to each other.');
+		}
+
 		ow(choices, `${ApplicationCommandOptionTypeNames[this.type]} choices`, choicesPredicate);
 
 		for (const [label, value] of choices) this.addChoice(label, value);
 		return this;
 	}
 
+	/**
+	 * Marks the option as autocompletable
+	 * @param autocomplete If this option should be autocompletable
+	 */
+	public setAutocomplete<U extends boolean>(
+		autocomplete: U,
+	): U extends true ? Omit<this, 'addChoice' | 'addChoices'> : this {
+		// Assert that you actually passed a boolean
+		ow(autocomplete, 'autocomplete', ow.boolean);
+
+		if (autocomplete && Array.isArray(this.choices) && this.choices.length > 0) {
```

Comment:
```suggestion
		if (autocomplete && this.choices?.length > 0) {
```

Reply:
TS doesn't like this, complains about it being potentially undefined.